python_modules/dagster/dagster/_grpc/proxy_server.py

```diff
@@ -216,6 +245,7 @@ def StreamingExternalRepository(self, request, context):
         return self._streaming_query("StreamingExternalRepository", request, context)
 
     def Heartbeat(self, request, context):
+        self.__last_heartbeat_time = time.time()
         return self._query("Heartbeat", request, context)
```

Comment:
I'm pretty sure we do not need to pass this one through like the others right? since its a different heartbeat? (the proxy server already sends heartbeats to its child code servers)

Reply:
ah - yea that's a good point. we should be able to avoid the additional send.